src/ristretto/dalek_range_proof.rs

```diff
@@ -88,6 +93,54 @@ impl RangeProofService for DalekRangeProofService {
+            .map_err(|_| RangeProofError::ProofConstructionError)?;
+        Ok(proof.to_bytes())
+    }
+
+    fn rewind_proof_value_only(&self, proof: &Self::P, commitment: &PedersenCommitment, rewind_public_key: &RistrettoPublicKey, rewind_blinding_public_key: &RistrettoPublicKey) -> Result<RewindResult, RangeProofError> {
+        let rp = DalekProof::from_bytes(&proof).map_err(|_| RangeProofError::InvalidProof)?;
+
+        let mut pt = Transcript::new(b"tari");
+        let rewind_nonce_1 = get_rewind_nonce_from_pub_key(&rewind_public_key.compressed, &commitment.as_public_key().compressed);
+        let rewind_nonce_2 = get_rewind_nonce_from_pub_key(&rewind_blinding_public_key.compressed, &commitment.as_public_key().compressed);
+        let (confidential_value, proof_message) = rp.rewind_single_get_value_only(&self.bp_gens, &mut pt, &commitment.as_public_key().compressed, self.range,&rewind_nonce_1,&rewind_nonce_2).map_err(|_| RangeProofError::ProofConstructionError)?;
+        if &proof_message[..REWIND_CHECK_MESSAGE.len()] != REWIND_CHECK_MESSAGE {
+            return Err(RangeProofError::InvalidRewind);
+        }
+        let mut truncated_proof_message: [u8; REWIND_USER_MESSAGE_LENGTH] = [0u8; REWIND_USER_MESSAGE_LENGTH];
+        truncated_proof_message.copy_from_slice(&proof_message[REWIND_CHECK_MESSAGE.len()..]);
+        Ok(RewindResult::new(confidential_value, truncated_proof_message))
+    }
+
+    fn rewind_proof_commitment_data(&self, proof: &Self::P, commitment: &PedersenCommitment, rewind_key: &RistrettoSecretKey, rewind_blinding_key: &RistrettoSecretKey) -> Result<FullRewindResult<RistrettoSecretKey>, RangeProofError> {
+        let rp = DalekProof::from_bytes(&proof).map_err(|_| RangeProofError::InvalidProof)?;
+
+        let mut pt = Transcript::new(b"tari");
+        let rewind_public_key = RistrettoPublicKey::from_secret_key(rewind_key);
+        let rewind_blinding_public_key = RistrettoPublicKey::from_secret_key(rewind_blinding_key);
+        let rewind_nonce_1 = get_rewind_nonce_from_pub_key(&rewind_public_key.compressed, &commitment.as_public_key().compressed);
+        let rewind_nonce_2 = get_rewind_nonce_from_pub_key(&rewind_blinding_public_key.compressed, &commitment.as_public_key().compressed);
+        let blinding_nonce_1 = get_secret_nonce_from_pvt_key(&rewind_key.0, &commitment.as_public_key().compressed);
+        let blinding_nonce_2 = get_secret_nonce_from_pvt_key(&rewind_blinding_key.0, &commitment.as_public_key().compressed);
+        let (confidential_value, blinding_factor, proof_message) = rp.rewind_single_get_commitment_data(&self.bp_gens,&self.pc_gens, &mut pt, &commitment.as_public_key().compressed, self.range,&rewind_nonce_1,&rewind_nonce_2,&blinding_nonce_1, &blinding_nonce_2).map_err(|_| RangeProofError::InvalidRewind)?;
+        if &proof_message[..REWIND_CHECK_MESSAGE.len()] != REWIND_CHECK_MESSAGE {
+            return Err(RangeProofError::InvalidRewind);
+        }
```

Comment:
This check is unnecessary as the  `rewind_single_get_commitment_data` will throw an error if the Bulletproof is not fully rewindable. It cannot return any values if all is not correct.